app/storyedit/storyedit.component.ts

```diff
@@ -62,37 +58,75 @@ export class StoryEditComponent implements OnInit, OnDestroy, CanDeactivate<CanD
   ngOnInit() {
     this.routeSub = this.route.params.subscribe(params => {
       this.id = +params['id'];
+      this.seriesId = +params['series_id'];
+
+      // must explicitly set the base-link for this edit/create route
+      if (this.id) {
+        this.editLink = ['/edit', this.id];
+      } else {
+        this.editLink = ['/create'];
+        if (this.seriesId) {
+          this.editLink.push(this.seriesId);
+        }
+      }
+
+      // (1) existing story, (2) new series-story, (3) new standalone-story
+      let auth = this.cms.follow('prx:authorization');
       if (this.id) {
-        this.cms.follow('prx:authorization').follow('prx:story', {id: this.id}).subscribe(doc => {
-          this.story = new StoryModel(null, doc);
+        auth.follow('prx:story', {id: this.id}).subscribe(storyDoc => {
+          if (storyDoc['appVersion'] !== 'v4') {
+            let oldLink = `https://www.prx.org/pieces/${this.id}`;
+            this.modal.alert(
+              'Cannot Edit Story',
+              `This episode was created in the older PRX.org app, and must be
+              edited there. <a target="_blank" href="${oldLink}">Click here</a> to view it.`,
+              () => { window.history.back(); }
+            );
+          } else if (storyDoc.has('prx:series')) {
+            storyDoc.follow('prx:series').subscribe(seriesDoc => {
+              this.story = new StoryModel(seriesDoc, storyDoc);
+              this.tab.setStory(this.story);
+            });
+          } else if (storyDoc.has('prx:account')) {
+            storyDoc.follow('prx:account').subscribe(accountDoc => {
+              this.story = new StoryModel(accountDoc, storyDoc);
+              this.tab.setStory(this.story);
+            });
+          } else {
+            console.error('WOH: story has no series or account!');
+            this.story = new StoryModel(null, storyDoc);
+            this.tab.setStory(this.story);
+          }
+        });
+      } else if (this.seriesId) {
+        auth.follow('prx:series', {id: this.seriesId}).subscribe(seriesDoc => {
+          this.story = new StoryModel(seriesDoc, null);
           this.tab.setStory(this.story);
         });
       } else {
-        this.cms.account.subscribe(accountDoc => {
+        auth.follow('prx:default-account').subscribe(accountDoc => {
           this.story = new StoryModel(accountDoc, null);
           this.tab.setStory(this.story);
         });
       }
     });
-    this.tabSub = this.tab.heroText.subscribe((text) => {
-      this.stepText = text;
-    });
   }
 
   ngOnDestroy() {
     this.routeSub.unsubscribe();
-    this.tabSub.unsubscribe();
   }
 
-  canDeactivate(next: any, prev: any): Observable<boolean> {
-    if (this.story.changed()) {
+  canDeactivate(next: any, prev: any): boolean | Observable<boolean> {
```

Comment:
Just a way to tell typescript that it'll return "one of those two things".  When you use it with an input param, typescript is also smart enough to recognize any `if (thing instanceof Something)` and cast it accordingly.